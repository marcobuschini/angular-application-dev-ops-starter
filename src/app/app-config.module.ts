import { NgModule, InjectionToken } from '@angular/core'
import { environment } from '../environments/environment'

export const APP_CONFIG = new InjectionToken<AppConfig>('app.config')

export class AppConfig {
  public apiEndpoint: string
  public googleId: string
}

export const APP_DI_CONFIG: AppConfig = {
  apiEndpoint: environment.apiEndpoint,
  googleId: environment.googleId,
}

export const AVAILABLE_LANGUAGES = new Map<string, string>()
AVAILABLE_LANGUAGES.set('en', 'English')
AVAILABLE_LANGUAGES.set('it', 'Italiano')

export const DEFAULT_LANGUAGE = 'en'

@NgModule({
  providers: [
    {
      provide: APP_CONFIG,
      useValue: APP_DI_CONFIG,
    },
  ],
})
export class AppConfigModule {}
