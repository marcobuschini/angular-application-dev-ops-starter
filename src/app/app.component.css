app-hexagon {
  position: absolute;
}

mat-sidenav {
  width: 20em;
}

mat-sidenav-container {
  height: calc(100% - 64px);
}

mat-sidenav mat-toolbar {
  margin-top: -64px;
}

mat-tree-node:focus {
  outline: none;
}

mat-tree-node {
  cursor: pointer;
}

mat-toolbar.top {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

mat-sidenav.left mat-toolbar {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}

mat-sidenav.right mat-toolbar {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}
