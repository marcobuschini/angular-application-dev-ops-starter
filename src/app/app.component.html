<mat-card>
  <mat-card-title>Welcome, developers!</mat-card-title>
  <mat-card-content
    >Hi, I'm here to help you live memorable experiences!</mat-card-content
  >
  <mat-card-actions>
    <button mat-button mat-flat-button color="primary" (click)="ok()">
      Ok
    </button>
    <button mat-button mat-flat-button color="accent" (click)="cancel()">
      Cancel
    </button>
  </mat-card-actions>
</mat-card>
