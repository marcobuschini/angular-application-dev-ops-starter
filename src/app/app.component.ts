import { Component } from '@angular/core'
import { MatCardModule } from '@angular/material/card'
import { MatButtonModule } from '@angular/material/button'
import {
  MatSnackBar,
  MatSnackBarConfig,
  MatSnackBarModule,
} from '@angular/material/snack-bar'

@Component({
  selector: 'app-root',
  styleUrls: ['./app.component.css'],
  templateUrl: './app.component.html',
  imports: [MatCardModule, MatButtonModule, MatSnackBarModule],
})
export class AppComponent {
  public title = 'frontend'
  public constructor(public snackbar: MatSnackBar) {}

  public ok(): void {
    this.snackbar.open('You clicked Ok', null, {
      duration: 1500,
    } as MatSnackBarConfig)
  }

  public cancel(): void {
    this.snackbar.open('You clicked Cancel', null, {
      duration: 1500,
    } as MatSnackBarConfig)
  }
}
