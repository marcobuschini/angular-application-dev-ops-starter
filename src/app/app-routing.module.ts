import {
  RouterModule,
  Routes,
  provideRouter,
  withComponentInputBinding,
} from '@angular/router'
import { NgModule } from '@angular/core'
import { AppComponent } from './app.component'

const routes: Routes = [
  { path: '', redirectTo: '/home', pathMatch: 'full' },
  { path: 'home', component: AppComponent },
]

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
  providers: [provideRouter(routes, withComponentInputBinding())],
})
export class AppRoutingModule {}
