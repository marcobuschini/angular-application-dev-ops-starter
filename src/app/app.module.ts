import { AppRoutingModule } from '../app/app-routing.module'
import { AppComponent } from './app.component'

import { AppConfigModule } from './app-config.module'

import { BrowserModule } from '@angular/platform-browser'
import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
import { NgModule } from '@angular/core'

@NgModule({
  declarations: [AppComponent],
  bootstrap: [AppComponent],
  imports: [
    BrowserModule,
    AppRoutingModule,
    AppConfigModule,
    BrowserAnimationsModule,
  ],
  providers: [],
})
export class AppModule {}
