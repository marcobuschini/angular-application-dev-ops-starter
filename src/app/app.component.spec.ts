import { TestBed } from '@angular/core/testing'
import { HttpTestingController } from '@angular/common/http/testing'
import { AppComponent } from './app.component'

import { NoopAnimationsModule } from '@angular/platform-browser/animations'

import { NGXLogger } from 'ngx-logger'
import { NGXLoggerMock } from 'ngx-logger/testing'
import { APP_CONFIG, AppConfig } from './app-config.module'

const TEST_DIRECTIVES = [AppComponent]

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [NoopAnimationsModule],
      declarations: TEST_DIRECTIVES,
      providers: [
        { provide: NGXLogger, useClass: NGXLoggerMock },
        { provide: APP_CONFIG, useClass: AppConfig },
        HttpTestingController,
      ],
    }).compileComponents()
    TestBed.inject(HttpTestingController)
  })

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent)
    const app = fixture.componentInstance
    expect(app).toBeTruthy()
  })

  it('should have as title "frontend"', () => {
    const fixture = TestBed.createComponent(AppComponent)
    const app = fixture.componentInstance
    expect(app.title).toEqual('frontend')
  })

  it('should render properly (not logged in)', () => {
    const fixture = TestBed.createComponent(AppComponent)
    fixture.detectChanges()
    const compiled = fixture.nativeElement
    expect(compiled).toMatchSnapshot()
  })
})
