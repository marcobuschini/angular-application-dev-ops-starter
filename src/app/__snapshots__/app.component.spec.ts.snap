// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render properly (not logged in) 1`] = `
<div
  id="root2"
>
  Hello frontend!

</div>
`;
